tests/Doctrine/Tests/ORM/Cache/DefaultRegionTest.php

```diff
@@ -99,3 +96,52 @@ public function testGetMulti()
         $this->assertEquals($value2, $actual[1]);
     }
 }
+
+final class SharedArrayCache extends ArrayCache
```

Comment:
This is super-weird... Could you explain the purpose of this class in a comment, please? (yes, I know that we discussed it in person) Also, I'd strongly suggest splitting out the file